cdap-examples/LogAnalysis/src/main/java/co/cask/cdap/examples/loganalysis/LogAnalysisApp.java

```diff
@@ -165,4 +177,80 @@ public void centers(HttpServiceRequest request, HttpServiceResponder responder,
+   */
+  public static final class RequestCounterHanlder extends AbstractHttpServiceHandler {
+
+    private static final Gson GSON = new Gson();
+    static final String REQUEST_COUNTER_PARTITIONS_HANDLER = "reqcount";
+    static final String REQUEST_FILE_CONTENT_HANDLER = "reqfile";
+    static final String REQUEST_FILE_PATH_HANDLER_KEY = "time";
+
+    @UseDataSet(REQ_COUNT_STORE)
+    private TimePartitionedFileSet reqCountStore;
+
+    /**
+     * Handler which lists all the different time partitions available in the {@link LogAnalysisApp#REQ_COUNT_STORE}
+     * {@link TimePartitionedFileSet}
+     */
+    @Path(REQUEST_COUNTER_PARTITIONS_HANDLER)
+    @GET
+    public void getRequestFilesetPartitions(HttpServiceRequest request, HttpServiceResponder responder) {
+
+      // get all the existing paritions
+      Set<TimePartitionDetail> partitionsByTime = reqCountStore.getPartitionsByTime(0, Long.MAX_VALUE);
+
+      if (partitionsByTime.isEmpty()) {
+        responder.sendString(HttpURLConnection.HTTP_NO_CONTENT, "No request count information found", Charsets.UTF_8);
+      } else {
+        Set<String> formatedTimes = Sets.newHashSet();
+        for (TimePartitionDetail timePartitionDetail : partitionsByTime) {
+          String partitionTime = DateFormat.getDateTimeInstance(DateFormat.SHORT, DateFormat.SHORT).format(
+            new Date(timePartitionDetail.getTime()));
+          formatedTimes.add(partitionTime);
+        }
+        responder.sendJson(HttpURLConnection.HTTP_OK, formatedTimes);
+      }
+    }
+
+    /**
+     * Handler which reads all the parts files from a given partition in {@link LogAnalysisApp#REQ_COUNT_STORE}
+     * {@link TimePartitionedFileSet} and send it as a string.
+     * Note: We make an assumption here that the contents for partitions in the tpfs for this example is not very huge.
+     * This method of serving contents is not ideal for large contents.
+     */
+    @Path(REQUEST_FILE_CONTENT_HANDLER)
+    @POST
+    public void getRequestFilesetContents(HttpServiceRequest request, HttpServiceResponder responder) {
+
+      String partition = GSON.fromJson(Charsets.UTF_8.decode(request.getContent()).toString(),
+                                 JsonObject.class).get(REQUEST_FILE_PATH_HANDLER_KEY).getAsString();
+      long partitionKey = 0;
+      try {
+        partitionKey = DateFormat.getDateTimeInstance(DateFormat.SHORT, DateFormat.SHORT).parse(partition).getTime();
+      } catch (ParseException e) {
+        responder.sendError(HttpURLConnection.HTTP_BAD_REQUEST, "Failed to parse the given string to a timestamp");
+      }
+
+      final Location location = reqCountStore.getPartitionByTime(partitionKey).getLocation();
+      if (location == null) {
+        responder.sendError(HttpURLConnection.HTTP_NO_CONTENT, "No files for the given date time string");
+      }
+
+      StringBuilder contents = new StringBuilder();
```

Comment:
why build this string in memory? Can't you write directly to the responder's channel?